src/main/java/org/jetbrains/id/names/suggesting/IdNamesSuggestingService.java

```diff
@@ -19,31 +20,54 @@ public static IdNamesSuggestingService getInstance(@NotNull Project project) {
     }
 
     public LinkedHashMap<String, Double> suggestVariableName(@NotNull PsiVariable variable) {
+        return suggestVariableName(variable, true);
+    }
+
+    public LinkedHashMap<String, Double> suggestVariableName(@NotNull PsiVariable variable,
+                                                             boolean showNotifications) {
         Instant timerStart = Instant.now();
         List<Prediction> nameSuggestions = new ArrayList<>();
         StringBuilder notifications = new StringBuilder();
+        Instant start = Instant.now();
+        double p = getVariableNameProbability(variable);
+        Instant end = Instant.now();
+        notifications.append(String.format("p=%.3f; t=%.3fms.\n",
+                p,
+                Duration.between(start, end).toNanos() / 1_000_000.));
         int prioritiesSum = 0;
         for (final VariableNamesContributor modelContributor : VariableNamesContributor.EP_NAME.getExtensions()) {
-            Instant start = Instant.now();
-            modelContributor.contribute(variable, nameSuggestions);
-            if (!nameSuggestions.isEmpty()) {
-                prioritiesSum += nameSuggestions.get(nameSuggestions.size() - 1).getPriority();
+            start = Instant.now();
+            prioritiesSum += modelContributor.contribute(variable, nameSuggestions);
+            end = Instant.now();
+            if (showNotifications) {
+                notifications.append(String.format("%s : %.3fms.\n",
+                        modelContributor.getClass().getSimpleName(),
+                        Duration.between(start, end).toNanos() / 1_000_000.));
             }
-            Instant end = Instant.now();
-            notifications.append(String.format("%s : %.3fms.\n",
-                    modelContributor.getClass().getSimpleName(),
-                    Duration.between(start, end).toNanos() / 1_000_000.));
         }
         LinkedHashMap<String, Double> result = rankSuggestions(nameSuggestions, prioritiesSum);
         Instant timerEnd = Instant.now();
-        notifications.append(String.format("Total time: %.3fms.\n",
-                Duration.between(timerStart, timerEnd).toNanos() / 1_000_000.));
-        NotificationsUtil.notify(variable.getProject(),
-                "Time of contribution:",
-                notifications.toString());
+        if (showNotifications) {
+            notifications.append(String.format("Total time: %.3fms.\n",
+                    Duration.between(timerStart, timerEnd).toNanos() / 1_000_000.));
+            NotificationsUtil.notify(variable.getProject(),
+                    "Time of contribution:",
+                    notifications.toString());
+        }
         return result;
     }
 
+    public Double getVariableNameProbability(@NotNull PsiVariable variable) {
+        double nameProbability = 0.0;
+        int prioritiesSum = 0;
```

Comment:
I prefer sum for now because with it we naturally get probability for every identifier, but with max we will have to normalize it in the end or just leave it as a confidence. I would like to see metrics that compare model with sum and with max. If sum model doesn't get any advantage over max model metrics wise than it would be obviously unnecessary overcomplication, but for now i would leave it as it is. Overflow problem might be solved with replacing to the `long` type.